src/db/data_source.h

```diff
@@ -0,0 +1,43 @@
+#ifndef SRC_DB_DATA_SOURCE_H_
+
+#define SRC_DB_DATA_SOURCE_H_
+
+#include <string>
+
+#include "absl/status/status.h"
+
+namespace db {
+
+// Abstract Database Interface that supports transaction operations.
+// TODO(n-triple-a): Implement interface using LMDB primitives
+class DataSource {
+ public:
+  DataSource() = default;
+  ~DataSource() = default;
+  
+  // Connects to the database instance.
+  virtual absl::Status Connect();
```

Comment:
Can we do `Connection` within the constructor which simplifies the interface for now?

Reply:
Yes that makes sense. Let me change that in the follow up PR.